tests/integration/helpers.py

```diff
@@ -121,3 +100,88 @@ async def perform_temporal_integrations(ops_test: OpsTest):
     )
 
     assert ops_test.model.applications[APP_NAME_SERVER].units[0].workload_status == "active"
+
+
+async def scale(ops_test: OpsTest, app, units):
+    """Scale the application to the provided number and wait for idle.
+
+    Args:
+        ops_test: PyTest object.
+        app: Application to be scaled.
+        units: Number of units required.
+    """
+    await ops_test.model.applications[app].scale(scale=units)
+
+    # Wait for model to settle
+    await ops_test.model.wait_for_idle(
+        apps=[app],
+        status="active",
+        idle_period=30,
+        raise_on_error=False,
+        raise_on_blocked=True,
+        timeout=300,
+        wait_for_exact_units=units,
+    )
+
+    assert len(ops_test.model.applications[app].units) == units
+
+
+async def setup_temporal_ecosystem(ops_test: OpsTest):
+    """Scale the application to the provided number and wait for idle.
+
+    Args:
+        ops_test: PyTest object.
+    """
+    await ops_test.model.deploy(APP_NAME_SERVER, channel="edge")
```

Comment:
What does the `await` wait for here? For an API call to be made or for the charm to actually get deployed? If the latter, it might be possible to call the four functions first and then wait for them later. Which should parallelize the wait and speed up the test.

Reply:
Updated to use `asyncio.gather` to group them together. Thanks!